src/js/actions/transform.js

```diff
@@ -533,6 +538,9 @@ define(function (require, exports) {
             })
             .then(function () {
                 return this.transfer(layerActions.resetIndex, document);
+            })
+            .then(function () {
```

Comment:
My gut says that the transaction should be ended prior to calling resetIndex ... but I don't actually know that it matters.